magma/ref.py

```diff
@@ -196,8 +196,9 @@ def set_defn(self, defn):
 
 
 class DerivedRef(Ref):
-    def __init__(self, parent):
+    def __init__(self, parent, index):
         self._parent = parent
+        self._index = index
```

Comment:
yes but all sub-classes of `DerivedRef` already have an `index` member right? (just being pedantic here, not a big deal).

Reply:
The code checks DerivedRef, so it's better to enforce the interface in the base class init then to expect that all subclasses happen to add a specific member